### PR TITLE
Bump kubernetes deps to v1.29.0

### PR DESCRIPTION
**Pull request by liangyuanpeng**
Fixes https://github.com/kubernetes-sigs/custom-metrics-apiserver/issues/160

If I understand correctly then it has been included in v1.29.0.

Note: go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp have update to v0.44.0

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,20 +1,22 @@
 module sigs.k8s.io/custom-metrics-apiserver
 
-go 1.20
+go 1.21
+
+toolchain go1.21.5
```
Comment (JorTurFer) [inline go.mod:5]: Do we need to set the toolchain?

**Review by JorTurFer — COMMENTED** [review]
There are compiling issues due to API changes (just updating the go.mod/go.sum isn't enough)

**Review by zroubalik — COMMENTED** [review]
Let's merge this https://github.com/kubernetes-sigs/custom-metrics-apiserver/pull/163 first, so we have updated k8s v1.28.5 deps.

File: go.mod (hunk 1 above)
Comment (liangyuanpeng) [reply go.mod:5]: Thanks for review, this is automatically generated when I use go mod tidy, let me check it.

Comment (JorTurFer) [reply go.mod:5]:
```
make test
```
will fail during the building process

Comment (liangyuanpeng) [reply go.mod:5]: Yes, have some breaking change on https://github.com/kubernetes-sigs/custom-metrics-apiserver/blob/261c3dd161e995a5bffb0e095c8d58e47edc2686/pkg/cmd/options/options.go#L102 , so i make this to draft.

**Comment by RainbowMango** [comment]
Hi @liangyuanpeng 
Can you help to rebase this PR and resolve the conflicts?

Karmada need this PR when bump Kubernetes deps to v1.29.

File: go.mod [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
 	github.com/stretchr/testify v1.8.4
-	k8s.io/api v0.28.1
-	k8s.io/apimachinery v0.28.1
-	k8s.io/apiserver v0.28.1
-	k8s.io/client-go v0.28.1
-	k8s.io/component-base v0.28.1
-	k8s.io/klog/v2 v2.100.1
-	k8s.io/kube-openapi v0.0.0-20230901164831-6c774f458599
-	k8s.io/metrics v0.28.1
+	k8s.io/api v0.29.0
```
Comment (RainbowMango) [inline go.mod:12]: We can go with v0.29.2 now.

**Comment by liangyuanpeng** [comment]
Sorry, I forgot about this after a vacation, I don't have the bandwidth to spare on this PR at the moment. :(

**Pull request closed without merging**